index.js

```diff
@@ -134,36 +146,43 @@ exports.registerTriggeredMethods = (componentName, stateMachineName, triggeredMe
     }
 }
 
-exports.startEventQueue = (configuration) => {
-    console.log('Registered triggered methods: ', triggeredMethods);
-
-    const installEventQueue = () => {
-        setInterval(eventQueue, 1000);
-        console.log('Waiting for tasks...');
-    };
-
+function updateConfiguration(configuration, callback) {
     if (configuration) {
         const postConfigurationAction = () => postConfiguration(configuration, configurationCallback);
-        const retryIntervalSeconds = 5;
         let configurationTimerID = null;
         const configurationCallback = (error) => {
             if (error) {
                 console.error('Error updating configuration: ', error);
-                console.error('Retrying in ', retryIntervalSeconds, 's...');
-                configurationTimerID = setTimeout(postConfigurationAction, retryIntervalSeconds * 1000);
+                callback(error, null);
                 return;
             }
-
-            clearTimeout(configurationTimerID);
             console.log('Configuraton successfuly updated!');
-            installEventQueue();
+            callback(null, true);
         };
 
         postConfigurationAction();
-        configurationTimerID = setTimeout(postConfigurationAction, retryIntervalSeconds * 1000);
+        installEventQueue(callback)

```

Comment:
do not install the queue until you got a success on the postConfiguration!

I am thinking about something like that:
```

function updateConfiguration(configuration, callback) {
    if (configuration) {
        const configurationCallback = (error) => {
            if (error) {
                console.error('Error updating configuration: ', error);
                callback && callback(error, null);
                return;
            }
            console.log('Configuraton successfuly updated!');
            installEventQueue(callback)
        };

        postConfiguration(configuration, configurationCallback);
    } else {
        installEventQueue(callback)
    }
}

```